File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let left = 0
    let minLength = Infinity
    let curSum =  0

    for(let i = 0; i < nums.length; i++){
        curSum += nums[i]
        while(curSum >= target){
            curSum -= nums[left]
            let len = (i - left) + 1
            minLength = Math.min(minLength,len)
            left++
        }
    }
    return minLength < Infinity ? minLength : 0
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        minLength = float("inf")
        curSum = 0

        for r in range(len(nums)):
            curSum += nums[r]
            while curSum >= target:
                curSum -= nums[l]
                length = (r - l) + 1
                minLength = min(minLength,length)
                l += 1
        return minLength if minLength < float("inf") else 0